### DISCUSSION TITLE
Best way to set application-wide FontSize?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7539] [Q&A] [answered]
Hi,

I'd like to set an application-wide default font-size. What's the best way to do this?

Currently, this seems to work in the App.axaml:

        <Style Selector=":is(TemplatedControl)">
            <Setter Property="FontSize" Value="{Binding Sizing.AppFontSize}"/>
        </Style>

However, it sets everything and I have difficulty overriding it for certain controls. Is there a better way?

**Comment by maxkatz6** [comment]
Set FontSize on a Window. It will be inherited.

**Reply by Firuz2006** [reply]
But it doesn't work in NumericUpDown

**Reply by robloo** [reply]
@Firuz2006 if you are using the Fluent v1 theme the controls use a resource to set FontSize rather than inheriting it. This is just what was done in WinUI upstream. See: https://github.com/AvaloniaUI/Avalonia/discussions/7539#discussioncomment-2118718 to fix this.

**Reply by maxkatz6** [reply]
> But it doesn't work in NumericUpDown

FontSize setter should be removed from there then.

**Comment by kuiperzone** [comment]
Thanks. However, Window.FontSize is not picked up by buttons, edit/check boxes or menu items. In fact, only TextBlock seems to inherit Window.FontSize.

(Am on Linux, Fedora, Wayland)

**Reply by kuiperzone** [reply]
Is this a bug?

![image](https://user-images.githubusercontent.com/17156238/152667226-149272a4-9f89-4a7f-8d92-3377d398b2f5.png)

**Reply by maxkatz6** [reply]
Yeah, looks like I was wrong, and fluent theme isn't adapted for that.

**Comment by kuiperzone** [comment]
Cheers, any ideas on how one could change the application font size with Fluent?

**Comment by maxkatz6** [comment, marked as answer]
Overriding SystemControlForegroundBaseHighBrush resource will work.
Put 
```
<x:Double x:Key="ControlContentThemeFontSize">14</x:Double>
```
In global resource dictionary (in the App.xaml for example).

**Comment by kuiperzone** [comment]
I'll give it a shot. Thank you.

**Comment by Firuz2006** [comment]
thanks everyone
